pkg/db/run.go

```diff
@@ -403,13 +403,23 @@ type RunFindQuery struct {
 	//
 	// If it is nil or empty, it means "match any".
 	Status []KnitRunStatus
+
+	// match if run's updated time is equal or later than Since.
+	Since *string
+
+	// match if run's updated time is equal or earlier than UpdatedAt + Duration.
+	Duration *string
```

Comment:
```suggestion
	// match if run's updated time is equal or later than Since.
	Since *time.Time

	// match if run's updated time is equal or earlier than UpdatedAt + Duration.
	Duration *time.Duration
```

When you can use a meaningful type, you should use the type. See `Status` field. It is typed as `KnitRunStatus`, it comes from the query parameter as `Since` and `Duration`.

If a value is typed `string`, the type `string` itself DOES NOT guarantee the value is writtein in a valid time expression (RFC3339, for example). But, if it is typed as `time.Time`, the validity is clear. It is a time, doubtlessly.

When you passing a value in a just `string`, any facts (including "it is verified") are lost. When you receiving a just `string`, you are saying that "it can accepts any strings in any format"[^doc-comment].
[^doc-comment]: It is worse that the documentation comment does not say about the format.

Functions **are** used in a specific context, but they **can be** used any context. Saying "The string should be in RFC3339!" demands attentions from users of the function. When it receives `time.Time`, it is ok that the compiler says ok, and extra attentions of human is not needed.

And more. Database related feature is not responsible to verify *raw* values in user input. Request handers are responsible that. If values are verified and parsed in a handler, it is *strongly recommended* that the fact of verification is expressed by types.

If you know the string value is in a valid format, you have done verification or parsing. Passing a value in `string` means "do verify and parse once again", it is redundant obviously.